toggl/toggl-button
# Public web login

Please remember the [Contributing Guidelines](https://github.com/toggl/toggl-button/blob/master/.github/CONTRIBUTING.md) :heart:

## :star2: What does this PR do?

- Revamp the login process of Toggl Button
- Add a custom login page which is displayed automatically on install
- It is also shown if the popup view is clicked in a logged-out state
- The login is deferred to a toggl.com page

## :bug: Recommendations for testing

#### All changes should be tested across Chrome and Firefox.

- Test this PR in conjunction with https://github.com/toggl/project-firefly/pull/878
- https://github.com/toggl/project-firefly/pull/878 can be pushed to production to test the PR for the production API (since its a standalone login page, it can be live before the extension changes are) 
- Test Google login
- See setup instructions on keybase at
   ```shell
   cat /keybase/team/toggl.frontend/toggl-button/1367/info.md
   ```
- Remove the currently installed dev extension, and load it again
- The login page should open
- If you close it, you should be able to access it from the popup as well
- Proceed with the login
- You should land back on the extension eventually
- The login page should render the correct view for the button state, ie. only show login option when user is logged out

## :memo: Links to relevant issues or information

Closes #1367
